src/components/button/button-types.style.js

```diff
@@ -9,92 +9,93 @@ export function makeColors(color) {
+      background: ${colors.error};
+      border-color: transparent;
+      color: ${colors.white};
+      &:hover {
+        background: ${colors.destructive.hover};
+      }
+      `)}
 
-    ${isDisabled ? `
+      ${IF(isDisabled, `
       background: ${disabled.button};
       color: ${disabled.text};
-      &:hover {
-        background: ${disabled.button};
-        color: ${disabled.text};
-      }
-    ` : ''}` : ''}
-  `,
-  secondary: `
+        &:hover {
+          background: ${disabled.button};
+          border-color: ${disabled.button};
+          color: ${disabled.text};
+        }
+      `)}      
+    `,
+    secondary: `
+        background: transparent;
+        border-color: ${colors.primary};
+        color: ${colors.primary};
+        &:hover {
+          background: ${colors.secondary};
+          border-color: ${colors.secondary};
+         ${makeColorAndIconColor(colors.white)}
+        }
+  
+        ${IF(isDisabled, `
+          border-color: ${disabled.button};
+          color: ${disabled.text};
+          &:hover {
+            background: transparent
+            border-color: ${disabled.button};
+            ${makeColorAndIconColor(disabled.text)}
+          }
+        `)}
+    `,
+    tertiary: `
       background: transparent;
-      border-color: ${colors.primary};
+      border-color: transparent;
       color: ${colors.primary};
       &:hover {
-        background: ${colors.secondary};
-        border-color: ${colors.secondary};
-       ${makeColors(colors.white)}
+        ${makeColorAndIconColor(colors.secondary)}
       }
-
-      ${isDisabled ? `
-        border-color: ${disabled.button};
+  
+      ${IF(isDisabled, `
         color: ${disabled.text};
         &:hover {
-          background: transparent
-          border-color: ${disabled.button};
-          ${makeColors(disabled.text)}
+          ${makeColorAndIconColor(disabled.text)}
         }
-    ` : ''}
-  `,
-  tertiary: `
-    background: transparent;
-    border-color: transparent;
-    color: ${colors.primary};
-    &:hover {
-      ${makeColors(colors.secondary)}
-    }
-
-    ${isDisabled ? `
-      color: ${disabled.text};
+      `)}
+    `,
+    darkBackground: `
+      background: ${colors.white};
+      border-color: transparent;
+      color: ${colors.primary};
       &:hover {
-        ${makeColors(disabled.text)}
+        background: ${colors.secondary};
+        ${makeColorAndIconColor(colors.white)}
       }
-    ` : ''}
-  `,
-  darkBackground: `
-    background: ${colors.white};
-    border-color: transparent;
-    color: ${colors.primary};
-    &:hover {
-      background: ${colors.secondary};
-      ${makeColors(colors.white)}
-    }
-
-    ${isDisabled ? `
-      background: ${disabled.button};
-      color: ${disabled.text};
-      &:hover {
+  
+      ${IF(isDisabled, `
         background: ${disabled.button};
-        ${makeColors(disabled.text)}
-      }
-    ` : ''}
-  `
-});
+        color: ${disabled.text};
```

Comment:
this color is common to all disabled buttons, maybe we could have it directly in the main file?